x-pack/filebeat/input/azureeventhub/v2_migration.go

```diff
@@ -0,0 +1,329 @@
+	blobContainerClient containerClient
+	checkpointStore     checkpointer
+}
+
+// newMigrationAssistant creates a new migration assistant.
+func newMigrationAssistant(log *logp.Logger, consumerClient consumerClient, blobContainerClient containerClient, checkpointStore checkpointer) *migrationAssistant {
+	return &migrationAssistant{
+		log:                 log,
+		consumerClient:      consumerClient,
+		blobContainerClient: blobContainerClient,
+		checkpointStore:     checkpointStore,
+	}
+}
+
+// checkAndMigrate checks if the v1 checkpoint information for the partitions
+// exists and migrates it to v2 if it does.
+func (m *migrationAssistant) checkAndMigrate(ctx context.Context, eventHubConnectionString, eventHubName, consumerGroup string) error {
+	// Fetching event hub information
+	eventHubProperties, err := m.consumerClient.GetEventHubProperties(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to get event hub properties: %w", err)
+	}
+
+	m.log.Infow(
+		"event hub information",
+		"name", eventHubProperties.Name,
+		"created_on", eventHubProperties.CreatedOn,
+		"partition_ids", eventHubProperties.PartitionIDs,
+	)
+
+	// Parse the connection string to get FQDN.
+	connectionStringInfo, err := parseConnectionString(eventHubConnectionString)
+	if err != nil {
+		return fmt.Errorf("failed to parse connection string: %w", err)
+	}
+
+	blobs, err := m.listBlobs(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, partitionID := range eventHubProperties.PartitionIDs {
+		err = m.checkAndMigratePartition(ctx, blobs, partitionID, connectionStringInfo.FullyQualifiedNamespace, eventHubName, consumerGroup)
+		if err != nil {
+			return fmt.Errorf("failed to check and migrate partition: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// checkAndMigratePartition checks if the v1 checkpoint information for the
+// `partitionID` partition.
+func (m *migrationAssistant) checkAndMigratePartition(
+	ctx context.Context,
+	blobs map[string]bool,
+	partitionID,
+	fullyQualifiedNamespace,
+	eventHubName,
+	consumerGroup string) error {
+
+	// v2 checkpoint information path
+	// mbranca-general.servicebus.windows.net/sdh4552/$Default/checkpoint/0
+	blob := fmt.Sprintf("%s/%s/%s/checkpoint/%s", fullyQualifiedNamespace, eventHubName, consumerGroup, partitionID)
+
+	// Check if v2 checkpoint information exists
+	if _, ok := blobs[blob]; ok {
+		m.log.Infow(
+			"checkpoint v2 information for partition already exists, no migration needed",
+			"partitionID", partitionID,
+		)
+
+		return nil
+	}
+
+	// Check if v1 checkpoint information exists
+	if _, ok := blobs[partitionID]; !ok {
+		m.log.Infow(
+			"checkpoint v1 information for partition doesn't exist, no migration needed",
+			"partitionID", partitionID,
+		)
+
+		return nil
+	}
+
+	// Try downloading the checkpoint v1 information for the partition
+	cln := m.blobContainerClient.NewBlobClient(partitionID)
+
+	// 4KB buffer should be enough to read
+	// the checkpoint v1 information.
+	buff := [4000]byte{}
```

Comment:
nitpick:

```suggestion
	buff := [4096]byte{}
```

To match with the comment: 4KB